v24/src/main/java/com/foursoft/harness/kbl/v24/HasLocatedComponents.java

```diff
@@ -0,0 +1,62 @@
+/*-
+ * ========================LICENSE_START=================================
+ * kbl-v24
+ * %%
+ * Copyright (C) 2020 - 2022 4Soft GmbH
+ * %%
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * =========================LICENSE_END==================================
+ */
+package com.foursoft.harness.kbl.v24;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public interface HasLocatedComponents {
+
+    List<KblAccessoryOccurrence> getAccessoryOccurrences();
+
+    List<KblAssemblyPartOccurrence> getAssemblyPartOccurrences();
+
+    List<KblComponentBoxOccurrence> getComponentBoxOccurrences();
+
+    List<KblConnectorOccurrence> getConnectorOccurrences();
+
+    List<KblFixingOccurrence> getFixingOccurrences();
+
+    List<KblSpecialTerminalOccurrence> getSpecialTerminalOccurrences();
+
+    List<KblWireProtectionOccurrence> getWireProtectionOccurrences();
+
+    default List<LocatedComponent> getLocatedComponents() {
+        return Stream.of(getComponentBoxOccurrences().stream()

```

Comment:
Why here the connectors and above the box itself?

Reply:
I don't know. The interface locatedComponent is implemented by KblComponentBoxConnectorOccurrence and from the KBLContainer can only get through the KblComponentBoxConnectorOccurrence. Or am I missing something?

Reply:
Oh I see. This is implemented by the Harness and really only the getLocatedComponents is new. Wouldn't it be better to just add soley this method to the Harness? We do that already somewhere else if I remember correctly.

Reply:
This is already in the Hanress. I have written in the previous comment container but meant harness .

hmm. I do not know where.